src/pages/TokenDetails/index.tsx

```diff
@@ -91,28 +92,34 @@ export default function TokenDetails() {
   const isBlockedToken = tokenWarning?.canProceed === false
 
   const navigate = useNavigate()
-  const switchChains = useCallback(
-    (newChain: Chain) => {
-      const chainSegment = newChain.toLowerCase()
+  const navigateToTokenForChain = useCallback(
+    (chain: Chain) => {
+      const chainName = chain.toLowerCase()
+      const token = tokenQueryData?.project?.tokens.find((token) => token.chain === chain && token.address)
       if (isNative) {
-        navigate(`/tokens/${chainSegment}/NATIVE`)
-      } else {
-        tokenQueryData?.project?.tokens?.forEach((token) => {
-          if (token.chain === newChain && token.address) {
-            navigate(`/tokens/${chainSegment}/${token.address}`)
-          }
-        })
+        navigate(`/tokens/${chainName}/${NATIVE_CHAIN_ID}`)
+      } else if (token) {
+        navigate(`/tokens/${chainName}/${token.address}`)
       }
     },
     [isNative, navigate, tokenQueryData?.project?.tokens]
   )
-  useOnGlobalChainSwitch(switchChains)
+  useOnGlobalChainSwitch(navigateToTokenForChain)
+  const navigateToWidgetSelectedToken = useCallback(
+    (input: Currency | undefined, output: Currency | undefined) => {
+      const update = output || input
+      if (!token || input?.equals(token) || output?.equals(token) || !update) return
```

Comment:
```input?.equals(token) || output?.equals(token)```

I think these might be redundant with the !update at the end

Reply:
Not redundant - but I rearranged the conditionals in case it makes it clearer.